packages/salesforcedx-vscode-core/src/decorators/scratch-org-decorator.ts

```diff
@@ -5,50 +5,38 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import * as fs from 'fs';
-import * as os from 'os';
-import * as path from 'path';
+import { ConfigUtil, projectPaths } from '@salesforce/salesforcedx-utils-vscode';
+// import * as fs from 'fs';
 import { StatusBarAlignment, StatusBarItem, window, workspace } from 'vscode';
 import { nls } from '../messages';
-import { workspaceUtils } from '../util';
 
-const CONFIG_FILE = workspaceUtils.hasRootWorkspace()
-  ? path.join(
-      workspaceUtils.getRootWorkspacePath(),
-      '.sfdx',
-      'sfdx-config.json'
-    )
-  : path.join(os.homedir(), '.sfdx', 'sfdx-config.json');
+const CONFIG_FILE = projectPaths.sfdxProjectConfig();
 
 let statusBarItem: StatusBarItem;
 
-export function showOrg() {
+export async function showOrg() {
   if (!statusBarItem) {
     statusBarItem = window.createStatusBarItem(StatusBarAlignment.Left, 50);
     statusBarItem.tooltip = nls.localize('status_bar_open_org_tooltip');
     statusBarItem.command = 'sfdx.force.org.open';
     statusBarItem.show();
   }
-  displayDefaultUserName(CONFIG_FILE);
+  await displayDefaultUserName();
 }
 
 export function monitorOrgConfigChanges() {
   const watcher = workspace.createFileSystemWatcher(CONFIG_FILE);
   watcher.onDidChange(uri => {
-    displayDefaultUserName(uri.fsPath);
+    displayDefaultUserName().catch(err => console.error(err));
   });
   watcher.onDidCreate(uri => {
-    displayDefaultUserName(uri.fsPath);
+    displayDefaultUserName().catch(err => console.error(err));
   });
 }
 
-function displayDefaultUserName(configPath: string) {
-  fs.readFile(configPath, (err, data) => {
-    if (!err) {
-      const config = JSON.parse(data.toString());
-      if (config['defaultusername']) {
-        statusBarItem.text = `$(browser)`;
-      }
-    }
-  });
+async function displayDefaultUserName() {
+  const defaultUsernameOrAlias = await ConfigUtil.getDefaultUsernameOrAlias();
+  if (defaultUsernameOrAlias) {
```

Comment:
It will make the tests easier to combine the logic from 'showOrg' and the logic from this function displayDefaultUserName/displayBrowserIcon so that the statusBarItem only gets created when needed:

<img width="500" alt="2022-11-08_06-37-12" src="https://user-images.githubusercontent.com/46458081/200593263-85aa992f-ef34-4fe0-858b-0892b4b67739.png">